src/components/EditOrgsForm.vue

```diff
@@ -0,0 +1,207 @@
+      <label class="ml-1">Test Data</label>
+    </div>
+    <div>
+      <PvCheckbox v-model="localOrgData.demoData" binary />
+      <label class="ml-1">Demo Data</label>
+    </div>
+  </div>
+</template>
+<script setup>
+import { ref, defineProps, onMounted, watch } from 'vue';
+import { useAuthStore } from '@/store/auth';
+import { storeToRefs } from 'pinia';
+import { fetchDocById } from '@/helpers/query/utils';
+import { useQuery } from '@tanstack/vue-query';
+import _isEmpty from 'lodash/isEmpty';
+const props = defineProps({
+  orgType: { type: String, required: true },
+  orgId: { type: String, required: true },
+  editMode: { type: Boolean, default: true },
+});
+
+// +------------+
+// | Initialize |
+// +------------+
+const initialized = ref(false);
+const authStore = useAuthStore();
+const { roarfirekit } = storeToRefs(authStore);
+
+const emit = defineEmits(['modalClosed', 'update:orgData']);
+
+// +----------------------------+
+// | Query for existing orgData |
+// +----------------------------+
+const { data: serverOrgData } = useQuery({
+  queryKey: ['org', props.orgType, props.orgId],
+  queryFn: () => fetchDocById(props.orgType, props.orgId),
+  keepPreviousData: true,
+  enabled: initialized,
+  staleTime: 5 * 60 * 1000, // 5 minutes
+});
+
+// +-------------+
+// | Local State |
+// +-------------+
+const localOrgData = ref({
+  name: null,
+  abbreviation: null,
+  address: null,
+  ncesId: null,
+  tags: [],
+  testData: false,
+  demoData: false,
+});
+
+const setupOrgData = (orgData) => {
+  console.log('setupOrgData', orgData);
+  let org = {
+    name: orgData?.name ?? '',
+    abbreviation: orgData?.abbreviation ?? '',
+    address: orgData?.address ?? '',
+    ncesId: orgData?.ncesId ?? '',
+    tags: orgData?.tags ?? [],
+    testData: orgData?.testData ?? false,
+    demoData: orgData?.demoData ?? false,
+  };
+  localOrgData.value = org;
+};
+
+const setAddress = (place) => {
+  localOrgData.value.address = {
+    addressComponents: place.address_components || [],
+    formattedAddress: place.formatted_address,
+    googlePlacesId: place.place_id,
+    googleMapsUrl: place.url,
+  };
+};
+
+watch(
+  () => serverOrgData,
+  (orgData) => {
+    if (!_isEmpty(orgData)) {
+      setupOrgData(orgData.value);
+    }
+  },
+  { deep: true, immediate: false },
+);
+
+// +--------------------+
+// | Initialize firekit |
+// +--------------------+
+let unsubscribe;
+const init = () => {
+  if (unsubscribe) unsubscribe();
+  initialized.value = true;
+};
+
+unsubscribe = authStore.$subscribe(async (mutation, state) => {
+  if (state.roarfirekit?.restConfig) init();
+});
+
+onMounted(() => {
+  if (roarfirekit.value?.restConfig) init();
+  if (!_isEmpty(serverOrgData.value)) setupOrgData(serverOrgData.value);
+});
+
+// +---------------------+
+// | Handle update event |
+// +---------------------+
+watch(
+  () => localOrgData,
+  (orgData) => {
+    console.log('emitting new orgData', orgData);
```

Comment:
Are the various console logs intentional or accidental left overs?

Reply:
Agreed. We should remove the console logs after development.